cms/djangoapps/export_course_metadata/signals.py

```diff
@@ -0,0 +1,40 @@
+"""
+This file exports metadata about the course.
+Initially the export goes to S3,
+but the storage class could be used for other backends in the future.
+
+This data is initially being used by Braze Connected Content to include
+section highlights in emails, but may be used for other things in the future.
+"""
+
+import json
+import logging
+
+from django.core.files.base import ContentFile
+from django.dispatch import receiver
+from openedx.core.djangoapps.schedules.content_highlights import get_all_course_highlights
+from xmodule.modulestore.django import SignalHandler
+
+from .storage import course_metadata_export_storage
+from .toggles import EXPORT_COURSE_METADATA_FLAG
+
+log = logging.getLogger(__name__)
+
+
+@receiver(SignalHandler.course_published)
+def export_course_metadata(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    Export course metadata on course publish.
+
+    File format
+    '{"highlights": [["week1highlight1", "week1highlight2"], ["week1highlight1", "week1highlight2"], [], []]}'
+    To retrieve highlights for week1, you would need to convert to json, then do
+    course_metadata['highlights'][0]
```

Comment:
I'm not sure this will work with the way Braze consumes content. At their documentation here https://www.braze.com/docs/user_guide/personalization_and_dynamic_content/connected_content/local_connected_content_variables/, they say if the response is a string, they basically just dump it in the email rather than letting it be worked with and formatted. 
Do you know if it is possible to upload to S3 in a way that when the GET request from Braze is fired, JSON will be returned?

Reply:
Mmm, yeah like setting content-type of a `ContentFile` maybe? I dunno either.

Reply:
this wasn't in the django-storages docs but looking at the code it seems like adding .json to the filename might do it, which i've done https://github.com/jschneier/django-storages/blob/1.8/storages/backends/s3boto3.py#L476

Reply:
Sounds good to me. I bring it up as a bigger point as this is something we should test prior to releasing because if it doesn't work for the Braze templates, then we will need to rework this implementation (although I'm sure 95% of this will remain the same and it might just be updating the content type or something). Once this is shipped and we can validate data in S3, I'm happy to help out ensuring it is working in Braze too :)